File: codeforces/cf1535B_array_reodering.py
# uninhm
# https://codeforces.com/contest/1535/problem/B
# greedy

from math import gcd

t = int(input())

for i in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    a = list(filter(lambda x: x%2==0, a)) + list(filter(lambda x: x%2==1, a))

    ans = 0
    for i in range(n):
        for j in range(i+1, n):
            if gcd(a[i], 2*a[j]) > 1:
                ans += 1
    print(ans)

File: codeforces/cf1535B_array_reodering.hs
-- uninhm
-- https://codeforces.com/contest/1535/problem/B
-- greedy

import Control.Monad (replicateM_)
import Data.List (tails)

solve a = length [ True | (x:rest) <- tails a, y <- rest, gcd x (2*y) > 1 ]

main = do
  t <- read <$> getLine
  replicateM_ t $ do
    n <- getLine
    a <- map read . words <$> getLine
    print $ solve $ filter even a ++ filter odd a
